### PR TITLE
searchWithTermAndLocation

### PR DESCRIPTION
**Pull request by gkye**
added a function to accept a location parameter.

### CODE REVIEW

File: Yelp/Business.swift [hunk 1]
```diff
@@ -91,4 +91,9 @@ class Business: NSObject {
     class func searchWithTerm(term: String, sort: YelpSortMode?, categories: [String]?, deals: Bool?, completion: ([Business]!, NSError!) -> Void) -> Void {
         YelpClient.sharedInstance.searchWithTerm(term, sort: sort, categories: categories, deals: deals, completion: completion)
     }
+    class func searchWithTermAndLocation(term: String, location: String, completion: ([Business]!, NSError!) -> Void) {
+        
+        YelpClient.sharedInstance.searchWithTermAndLocation(term, location: location, completion: completion)
+    }
+    
 }
```
Comment (timothy1ee) [inline Yelp/Business.swift:99]: Can you fix the whitespace here? There should be a space between functions, and there shouldn't be a space in the first line of the function.

File: Yelp/YelpClient.swift [hunk 2]
```diff
@@ -57,10 +57,10 @@ class YelpClient: BDBOAuth1RequestOperationManager {
     
     func searchWithTerm(term: String, sort: YelpSortMode?, categories: [String]?, deals: Bool?, completion: ([Business]!, NSError!) -> Void) -> AFHTTPRequestOperation {
         // For additional parameters, see http://www.yelp.com/developers/documentation/v2/search_api
-
+        
         // Default the location to San Francisco
         var parameters: [String : AnyObject] = ["term": term, "ll": "37.785771,-122.406165"]
-
+        
         if sort != nil {
```
Comment (timothy1ee) [inline Yelp/YelpClient.swift:68]: Can you remove this extra whitespace?

File: Yelp/YelpClient.swift [hunk 3]
```diff
@@ -84,4 +84,26 @@ class YelpClient: BDBOAuth1RequestOperationManager {
...
+        return self.GET("search", parameters: parameters, success: { (operation: AFHTTPRequestOperation!, response: AnyObject!) -> Void in
+            let dictionaries = response["businesses"] as? [NSDictionary]
+            if dictionaries != nil {
+                completion(Business.businesses(array: dictionaries!), nil)
+            }
+            }, failure: { (operation: AFHTTPRequestOperation?, error: NSError!) -> Void in
+                completion(nil, error)
+        })!
+    }
 }
```
Comment (timothy1ee) [inline Yelp/YelpClient.swift:94]: If you're adding a convenience method, can you chain everything through the same method? That way there is less duplicate code and less extraneous code paths where bugs can crop up.

Also, can you remove any print statements that you added for testing?

File: Yelp/YelpClient.swift [hunk 4]
```diff
@@ -84,4 +84,26 @@ class YelpClient: BDBOAuth1RequestOperationManager {
...
+    
+    func searchWithTermAndLocation(term: String, location: String, completion: ([Business]!, NSError!) -> Void) -> AFHTTPRequestOperation {
+        // For additional parameters, see http://www.yelp.com/developers/documentation/v2/search_api
+        
+        // Default the location to San Francisco
+        var parameters: [String : AnyObject] = ["term": term]
+                
+        //change ll to location (ll return nil)
+        parameters["location"] = location;
+
```
Comment (timothy1ee) [inline Yelp/YelpClient.swift]: Note that the reason we used "ll" was that causes the results to return a distance from the restaurant. I believe that adding "location" doesn't return a distance. I could be wrong there, but that's what I remember right now.

**Comment by timothy1ee** [comment]
Thanks for the pull request! I added some feedback on your changes.

File: Yelp/YelpClient.swift (hunk 4 above)
Comment (gkye) [inline Yelp/YelpClient.swift]: You're correct, there's no distance return when the location parameters added.

**Comment by gkye** [comment]
Thanks for the great feedback! I've made some changes accordingly.

**Comment by timothy1ee** [comment]
Thanks, looks good. We have some students going through the homework using this repository right now. I'm going to test this a bit more before I merge.

Appreciate the pull request!

**Comment by gkye** [comment]
No problem!

**Pull request closed without merging**